test/test_torch.py

```diff
@@ -12151,6 +12151,25 @@ def test_ceil_out_mismatch(self, device):
         b = torch.randn(1, device=device)
         self.assertRaises(RuntimeError, lambda: torch.ceil(a, out=b))
 
+    # Test for unary ops appropriate for floating-type output from integral and boolean inputs
+    @dtypes(torch.int8, torch.int16, torch.int32, torch.int64, torch.bool)
+    def test_intbool_to_float_upcasting(self, device, dtype):
+        # Not all the functions are appropriate for this upcasting, so only test for the list below
+        op_list = ["acos", "asin", "ceil", "expm1", "floor", "log", "log10", "log1p", "log10", "log1p",
+                   "log2", "sin", "sinh", "sqrt", "trunc", "atan", "cos", "cosh", "exp", "tan",
+                   "tanh"]
+        # Make sure that the ops promote to floating default types
+        # Test both float32 and double (float64) dtypes
+        saved_dtype = torch.get_default_dtype()
+        default_floating_types = [torch.float32, torch.float64]
+        for default_floating_type in default_floating_types:
+            torch.set_default_dtype(default_floating_type)
+            for op_name in op_list:
+                x = torch.tensor([2], dtype=dtype, device=device)
+                out = getattr(torch, op_name)(x)
+                # Assert all int and bool dtypes are promoted to default floating dtype
+                self.assertIs(out.dtype, torch.get_default_dtype())
+        torch.set_default_dtype(saved_dtype)
```

Comment:
FYI there's a decorator `@float_double_default_dtype` which will run the test twice with each default dtype set. It uses a try/catch to ensure the original default is restored even if the particular test fails.

Reply:
Awesome. Thanks @nairbv, I didn't know of that decorator.